### DISCUSSION TITLE
EnsureChannelFirst after Lambda

### DISCUSSION THREAD

**Discussion by aeye-slater** [Project-MONAI/MONAI#7347] [General]
I was trying to implement a CLAHE transform using a Lambda Transform.

this is my normal transform for MONAI:
```
xform = mt.Compose(
        [
            mt.LoadImageD(keys=["image","mask"], image_only=False,reader='PILreader'),
            mt.EnsureChannelFirstD(["image","mask"]),
            mt.ResizeD(keys=["image","mask"], spatial_size=(512, 512)),
        ]
    )
```

If I add in the Lambda Transform I have to switch to the following:
```
xform = mt.Compose(
        [
            mt.LoadImageD(keys=["image","mask"], image_only=False,reader='PILreader'),
            mt.LambdaD(keys=["image"],func=clahe_monai),
            mt.EnsureChannelFirstD(["image"],channel_dim=-1),
            mt.EnsureChannelFirstD(["mask"]),
            mt.ResizeD(keys=["image","mask"], spatial_size=(512, 512)),       
        ]
    )
```
If I don't split the image and mask up for the ChannelFirst, I get the following error:
`ValueError: Unknown original_channel_dim in the MetaTensor meta dict or `meta_dict` or `channel_dim`.`

I checked this and the original_channel_dim is there for both (-1-> image nan->mask)  Somehow the lambda is causing an issue.

The only difference is adding the LambdaFunction, so clearly I don't understand something that is being passed from transform to transform as the CLAHE only changes the image part of the metatensor, but somehow kills the ability to EnsureCHannelFirst to work as it previously did.  Yes I have a fix, but it just didn't make sense.

Some guidance would be appreciated.

**Comment by KumoLiu** [comment]
Hi @aeye-slater, thanks for your interest here.
I guess the reason is that `clahe_monai` has changed the  `image` type. Perhaps you could add a `ToMetaTensor` at the end of the `clahe_monai`. 
Thanks!

**Reply by aeye-slater** [reply]
Ahhh thank you very much.  I figured it was something subtle, and yes I did have to change the type, I put it back to a tensor, but not a metatensor.
